Products/CMFPlone/browser/iconresolver.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+from OFS.Image import File
+from plone.registry.interfaces import IRegistry
+from Products.CMFCore.interfaces import ISiteRoot
+from zope.component import adapter
+from zope.component import getUtility
+from zope.component.hooks import getSite
+from zope.interface import implementer
+from zope.interface import Interface
+from zope.location.interfaces import LocationError
+from zope.traversing.interfaces import ITraversable
+from zope.publisher.interfaces import IPublishTraverse
+
+
+@implementer(IPublishTraverse)
+class IconBaseResolver:
+
+    def __init__(self, context, name):
+        self.context = context
+        self.name = name
+
+    def __call__(self):
+        registry = getUtility(IRegistry)
+        prefix = 'plone.staticresources.icon.'
+        icon = prefix + self.name
+        if icon in registry:
```

Comment:
```suggestion


PREFIX = 'plone.staticresources.icon.'

@implementer(IPublishTraverse)
class IconBaseResolver:

    def __init__(self, context, name):
        self.context = context
        self.name = name

    def __call__(self):
        registry = getUtility(IRegistry)
        icon = PREFIX + self.name
        if icon in registry:
```

Reply:
I would set the `PREFIX` as a class variable:
```
class IconBaseResolver:
     prefix = 'plone.staticresources.icon.'
```
IMO it is easier to override.

Reply:
ok, good point